src/Data/Rational/Properties.agda

```diff
@@ -1719,6 +1720,91 @@ toℚᵘ-homo-∣-∣ (mkℚ -[1+ _ ] _ _) = *≡* refl
 ∣∣p∣∣≡∣p∣ : ∀ p → ∣ ∣ p ∣ ∣ ≡ ∣ p ∣
 ∣∣p∣∣≡∣p∣ p = 0≤p⇒∣p∣≡p (0≤∣p∣ p)
 
+------------------------------------------------------------------------
+-- Metric space
+------------------------------------------------------------------------
+
+private
+  d : ℚ → ℚ → ℚ
+  d p q = ∣ p - q ∣
```

Comment:
This is really very frustrating. Anyone have any nice ideas how we can make this a function in `Data.Rational.Base`? I guess we could use some weird unicode bars?

Reply:
See my global comment: we should have `ℚ⁺` as a first class thing in the library before we proceed!

Reply:
Re: notation.
It *is* a norm, so does using the following offer any assistance?
```suggestion
-- distance function
∥_─_∥ : ℚ → ℚ → ℚ
∥ p ─ q ∥ = ∣ p - q ∣
```
(reusing the 'long minus' from `Data.List.Relation.Unary.Any`, whose Unicode codepoint/Agda input-encoding I confess not ever to be able to remember, so this might do more harm than good...). The "weird Unicode bars" here are in fact simply given by `\ | |`... so they aren't too bad, but the long minus is annoying...